tests/hamcrest_unit_test/core/raises_test.py

```diff
@@ -42,11 +41,11 @@ def testDoesNotMatchIfTheWrongExceptionTypeIsRaised(self):
         self.assert_does_not_match('Wrong exception',
                             raises(IOError),
                             calling(raise_exception))
-        expected_message = "AssertionError('(){}',) of type <%s> was raised instead" % \
+        expected_message = r"AssertionError\('\(\)\{\}',?\) of type <%s> was raised instead" % \
```

Comment:
Can you elaborate on what test was broken and how that manifested?

Reply:
On 3.6 and under, the exception's repr was `AssertionError('(){}',)`. In 3.7, the comma disappeared, so we were getting `AssertionError('(){}')`.

One option was to match against a regex, making the comma optional. That's what I went for. Another option would be to have 2 versions of the test, and use conditional skips. Happy to do that instead - it would be more code, but probably clearer. 

I wish I was making these changes with a pair to talk these things over with! ;-)

Reply:
I think since we have pytest's skipif mechanism to work with, I'd rather have a set of tests for each permutation, and select them accordingly. It's easier to follow.

Reply:
Okey dokey, I've split the test into 3 versions - one for Python 2, one for Python 3 up to 3.6, and one for 3.7 and up.